app/javascript/react/containers/CuisineShowContainer.js

```diff
@@ -1,14 +1,24 @@
 import React, { Component } from 'react';
 import RestaurantListTile from '../components/RestaurantListTile';
+import RestaurantFormContainer from '../containers/RestaurantFormContainer'
 
 class CuisineShowContainer extends Component {
   constructor(props) {
     super(props);
     this.state = {
       name: "",
       id: null,
-      restaurantsArray:[]
+      restaurantsArray:[],
+      restaurantsUpdated: false
     };
+    this.updateRestaurantsList = this.updateRestaurantsList.bind(this)
+
+  }
+
+  updateRestaurantsList(updated) {
+    let rests = this.state.restaurantsArray
+    rests[updated["restaurant"].id] = updated["restaurant"]
+    this.setState({restaurantsArray: rests})
```

Comment:
Something about this method is draining my soul. I have no context on how line 20 works based on reading it, and im not sure where this method is even being used

Reply:
This method updated the list of Restaurants on the page. We need to fit this updated restaurant back into the data structure kept by the parent component (i.e. Object.assign was not correctly copying the new restaurant into the restaurantsArray). This method on line 20 works and if functional